### ISSUE TITLE
don't answer html when client only accept application/json

### ISSUE THREAD

**Issue by bmillemathias** [OpenTSDB/opentsdb#1042]
Hello

When we came through the bug #324 we found out openTSDB answered with html while the client passed 'Accept: application/json', hence our client was not able to parse the output.
That would be sense that errors reported by openTSDB in the format expected by the client.

version: 2.2.1

**Comment by manolama** [comment]
Yeah we need to fix this for version 3.0. It's due to the backwards compatibility with 1.x's APIs.
